### PR TITLE
dlb: Add dpdk-demo yaml and Change IsSriovPfwithVFs function

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
Add dpdk-demo yaml
Remove Makefile

Change the function to use GetSriovNumVFs func instead of os.ReadFile that is already used in GetSriovNumVFs

Signed-off-by: Hyeongju Johannes Lee <hyeongju.lee@intel.com>

### CODE REVIEW

**Review by bart0sh — APPROVED** [review]
/lgtm

File: demo/dlb-dpdk-demo-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+spec:
+  restartPolicy: Never
+  containers:
+  - name: pf
+    image: intel/dlb-dpdk-demo:devel
+    imagePullPolicy: IfNotPresent
+    resources:
+      limits:
+        dlb.intel.com/pf: 1
+        cpu: 1
```
Comment (mythi) [inline demo/dlb-dpdk-demo-pod.yaml:14]: nice idea to combine both "demos" in one pod. 

I'd suggest that we make this Guaranteed pod (cpu/memory limits and requests are set and they're the same)

File: demo/dlb-dpdk-demo/Makefile [hunk 2]
```diff
@@ -1,4 +1,4 @@
 build:
```
Comment (bart0sh) [inline demo/dlb-dpdk-demo/Makefile:1]: I'd suggest to remove this file and check that image can be built with 'make dlb-dpdk-demo'.

File: demo/dlb-dpdk-demo/test.sh [hunk 3]
```diff
@@ -1,14 +1,15 @@
 #!/bin/sh
+dlb_dev=$(ls /dev/dlb* | sed 's/\/dev\/dlb//' | head -1)
 
 echo '\n1. Order queue test'
 echo '-------------------'
-dpdk-test-eventdev --no-huge -- --test=order_queue --nb_flows 64 --nb_pkts 512  --plcores 1 --wlcores 2-7
+dpdk-test-eventdev --no-huge --vdev="dlb2_event,dev_id=`echo $dlb_dev`" -- --test=order_queue --nb_flows 64 --nb_pkts 512  --plcores 1 --wlcores 2-7
```
Comment (bart0sh) [inline demo/dlb-dpdk-demo/test.sh:6]: Why do we need this and below changes?

File: demo/dlb-dpdk-demo/Makefile (hunk 2 above)
Comment (hj-johannes-lee) [reply demo/dlb-dpdk-demo/Makefile:1]: I see.!
Actually, I wanted to add the message while making a pr if I will just remove or not. Let me remove it.!

File: demo/dlb-dpdk-demo-pod.yaml (hunk 1 above)
Comment (hj-johannes-lee) [reply demo/dlb-dpdk-demo-pod.yaml:14]: Then, does same thing have to apply to the dlb-libdlb-demo-pod.yaml?

File: cmd/internal/pluginutils/sriov.go [hunk 4]
```diff
@@ -22,9 +22,9 @@ import (
 
 // IsSriovPFwithVFs returns true if the device with given pfpath has virtual functions, false otherwise.
 func IsSriovPFwithVFs(pfpath string) bool {
-	dat, err := os.ReadFile(path.Join(pfpath, "device/sriov_numvfs"))
+	dat := GetSriovNumVFs(pfpath)
 
-	if err == nil && strings.TrimSpace(string(dat)) != "0" {
+	if dat != "-1" && dat != "0" {
```
Comment (bart0sh) [inline cmd/internal/pluginutils/sriov.go:27]: why not ```return dat != "-1" && dat != "0"``` as I've initially suggested?

**Pull request merged**